addons/l10n_in_check_gst_status/models/res_partner.py

```diff
@@ -0,0 +1,108 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+import logging
+from odoo import api, fields, models, _
+from odoo.addons.iap import jsonrpc
+from odoo.exceptions import UserError, AccessError
+
+_logger = logging.getLogger(__name__)
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    l10n_in_gstin_verified_status = fields.Selection([
+            ("active", "Active"),
+            ("inactive", "Inactive"),
+            ("invalid", "Invalid"),
+        ],
+        compute="_compute_l10n_in_gst_status",
+        string="GST Status",
+        store=True,
+        tracking=True,
+    )
+    l10n_in_gstin_verified_date = fields.Date(
+        string="GSTIN Verified Date",
+        compute="_compute_l10n_in_gst_status",
+        store=True,
+        tracking=True,
+    )
+    l10n_in_gstin_status_color = fields.Selection([
+            ("green", "Green"),
+            ("yellow", "Yellow"),
+            ("red", "Red"),
+        ],
+        default='red',
+        compute="_compute_l10n_in_gstin_status_color",
+    )
+
+    def _l10n_in_gstin_status_get_url(self):
+        return "https://jva-odoo-iap-apps2-15-0-gst-number-active-check-jva-13181640.dev.odoo.com/iap/l10n_in_reports/1/public/search"
+
+    @api.depends('vat')
+    def _compute_l10n_in_gst_status(self):
+        """
+        Reset GST Status Whenever the `vat` of partner changes
+        """
+        for partner in self:
+            if partner.country_code == 'IN':
+                partner.l10n_in_gstin_verified_status = False
+                partner.l10n_in_gstin_verified_date = False
+
+    @api.depends('l10n_in_gstin_verified_status', 'l10n_in_gstin_verified_date')
+    def _compute_l10n_in_gstin_status_color(self):
+        for partner in self:
+            if partner.country_code == 'IN':
+                partner.l10n_in_gstin_status_color = 'red'
+                if partner.l10n_in_gstin_verified_status == 'active':
+                    diff = (fields.Date.today() - partner.l10n_in_gstin_verified_date).days
+                    if diff >= 0 and diff <= 30:
+                        partner.l10n_in_gstin_status_color = 'green'
+                    elif diff > 30 and diff <= 60:
+                        partner.l10n_in_gstin_status_color = 'yellow'
+            else:
+                partner.l10n_in_gstin_status_color = False
+
+    def l10n_in_get_gstin_verified_status(self):
+        self.ensure_one()
+        url = self._l10n_in_gstin_status_get_url()
```

Comment:
```suggestion
        self.ensure_one()
        if not self.vat:
            raise ValidationError(_(...message))
        url = self._l10n_in_gstin_status_get_url()
```
May be a validation error? if someone tries to call the method explicitly?